frontend/SideDish/SideDish/View/MainDiffableDataSource.swift

```diff
@@ -0,0 +1,50 @@
+
+import Foundation
+import UIKit
+
+class MainDiffableDataSource {
+    
+    enum sectionTitle: String, CaseIterable {
+        case main = "모두가 좋아하는 든든한 메인요리"
+        case soup = "정성이 담긴 뜨끈뜨끈 국물요리"
+        case side = "식탁을 풍성하게 하는 정갈한 밑반찬"
+    }
+    
+    
+    private var dataSource : UICollectionViewDiffableDataSource<sectionTitle, MenuViewModel>!
```

Comment:
1. 
```suggestion
    private var dataSource: UICollectionViewDiffableDataSource<sectionTitle, MenuViewModel>!
```

2. UICollectionViewDiffableDataSource<sectionTitle, MenuViewModel>! 표현하고 setupDataSource 함수를 만들 수도 있지만  `lazy var` 이용하여 생성할 수도 있습니다. 전자와 후자 방법으로 어떤 차이가 있을까요?

Reply:
둘다 초기에 값을 넣지 않는다는 공통점이 있지만 전자 처럼 IUO로 할 경우, optional값을 다루게 되어 이후에 직접 사용할 때마다 optional을 해제해야합니다. 하지만 lazy를 사용할 경우 첫 호출시 값을 설정해주어 이후에는 optional을 다루지 않는다는 차이가 있는 것 같습니다! 따라서 본래 optional을 다루는 프로퍼티라면 !으로 표현해도 괜찮겠지만, 그게 아니라면 lazy가 더 나을 것 같습니다!